test/conversations/Conversations.test.ts

```diff
@@ -148,4 +152,121 @@ describe('conversations', () => {
     expect(aliceConversationsList).toHaveLength(1)
     expect(bobConversationList).toHaveLength(1)
   })
+
+  describe('newConversation', () => {
+    it('uses an existing v1 conversation when one exists', async () => {
+      const aliceConvo = await alice.conversations.newConversation(bob.address)
+      expect(aliceConvo instanceof ConversationV1).toBeTruthy()
+      await aliceConvo.send('gm')
+      const bobConvo = await bob.conversations.newConversation(alice.address)
+      expect(bobConvo instanceof ConversationV1).toBeTruthy()
+    })
+
+    it('continues to use v1 conversation even after upgrading bundle', async () => {
+      const aliceConvo = await alice.conversations.newConversation(bob.address)
+      await aliceConvo.send('gm')
+      expect(aliceConvo instanceof ConversationV1).toBeTruthy()
+      await bob.publishUserContact(false)
+      await sleep(50)
+
+      const aliceConvo2 = await alice.conversations.newConversation(bob.address)
```

Comment:
I think if you want to make sure alice picks up the new bundle for bob, you have to make her forget the old one (at least I did in my test). It might be good to add assertions for that.

Reply:
https://github.com/xmtp/xmtp-js/blob/dcafb40098b5ae82519b67f3924aa97c98463f72/test/conversations/Conversation.test.ts#L153-L159